extras/minimize_testcase.cpp

```diff
@@ -0,0 +1,251 @@
+
+#include "polygon.h"
+#include "utils.h"
+
+using namespace manifold;
+
+bool safeToRemove(Polygons &polys, int i, int j, float precision) {
+  if (polys[i].size() == 3) return false;
+  glm::vec2 prev = polys[i][j == 0 ? polys[i].size() - 1 : (j - 1)];
+  glm::vec2 next = polys[i][j == (polys[i].size() - 1) ? 0 : (j + 1)];
+  for (int k = 0; k < polys.size(); k++) {
+    auto ll = [&](size_t l) {
+      return l == (polys[k].size() - 1) ? 0 : (l + 1);
+    };
+    glm::vec2 *polysk = polys[k].data();
+    if (!std::all_of(
+#if MANIFOLD_PAR == 'T' && !(__APPLE__)
+            std::execution::par,
+#endif
+            countAt(0lu), countAt(polys[k].size()), [=](size_t l) {
+              if (i == k && (l == j || ll(l) == j)) return true;
+              return !intersect(prev, next, polysk[l], polysk[ll(l)],
+                                precision);
+            }))
+      return false;
+  }
+  return true;
+}
+
+std::pair<int, int> findIndex(const Polygons &polys, int i) {
+  int outer = 0;
+  while (i >= polys[outer].size()) i -= polys[outer++].size();
+  return std::make_pair(outer, i);
+}
+
+void Dump(const Polygons &polys) {
+  for (auto poly : polys) {
+    std::cout << "polys.push_back({" << std::setprecision(9) << std::endl;
+    for (auto v : poly) {
+      std::cout << "    {" << v.x << ", " << v.y << "},  //" << std::endl;
+    }
+    std::cout << "});" << std::endl;
+  }
+  for (auto poly : polys) {
+    std::cout << "show(array([" << std::endl;
+    for (auto v : poly) {
+      std::cout << "  [" << v.x << ", " << v.y << "]," << std::endl;
+    }
+    std::cout << "]))" << std::endl;
+  }
+}
+
+void DumpTriangulation(const Polygons &polys, float precision) {
+  ExecutionParams oldParams = PolygonParams();
+  manifold::PolygonParams().processOverlaps = true;
+  auto result = Triangulate(polys);
+  PolygonParams() = oldParams;
+  for (auto &tri : result) {
+    auto xid = findIndex(polys, tri.x);
+    auto yid = findIndex(polys, tri.y);
+    auto zid = findIndex(polys, tri.z);
+    auto x = polys[xid.first][xid.second];
+    auto y = polys[yid.first][yid.second];
+    auto z = polys[zid.first][zid.second];
+    printf("show(array([[%.7f, %.7f], [%.7f, %.7f], [%.7f, %.7f]]))\n", x.x,
+           x.y, y.x, y.y, z.x, z.y);
+    if (!CCW(x, y, z, precision)) {
+      printf("^ not CCW\n");
+    }
+  }
+}
+
+// x direction ray
+// idk which direction, but it is a fixed direction...
+bool rayHit(glm::vec2 point, glm::vec2 q0, glm::vec2 q1) {
+  return ((q0.y >= point.y) != (q1.y >= point.y)) &&
+         (point.x <= (q1.x - q0.x) * (point.y - q0.y) / (q1.y - q0.y) + q0.x);
+}
+
+std::vector<int> getChildren(Polygons &polys, int i) {
+  std::vector<int> results;
+  for (int j = 0; j < polys.size(); j++) {
+    if (i == j) continue;
+    glm::vec2 point = polys[j][0];
+    auto k1 = [&](size_t k) {
+      return k == (polys[i].size() - 1) ? 0 : (k + 1);
+    };
+    int count = std::count_if(
+#if MANIFOLD_PAR == 'T' && !(__APPLE__)
```

Comment:
Can we use `__has_include` instead?

Reply:
I'd like to avoid `__APPLE__` and any other platform-specific macros if possible.